dbms/src/Storages/StorageHDFS.cpp

```diff
@@ -138,12 +187,33 @@ BlockInputStreams StorageHDFS::read(
     size_t max_block_size,
     unsigned /*num_streams*/)
 {
-    return {std::make_shared<HDFSBlockInputStream>(
-        uri,
-        format_name,
-        getSampleBlock(),
-        context_,
-        max_block_size)};
+    Strings path_parts;
+    size_t first_glob = uri.find_first_of("*?{");
```

Comment:
Need to generalize this code according to the recommendation in StorageFile. We shouldn't have two different cases -- with glob and without glob.